leakybucket_test.go

```diff
@@ -143,11 +144,64 @@ func (s *LimitersTestSuite) TestLeakyBucketOverflow() {
 	}
 }
 
+func (s *LimitersTestSuite) TestLeakyBucketNoExpiration() {
+	clock := l.NewSystemClock()
+	buckets := s.leakyBuckets(1, time.Minute, 0, clock)
+
+	// Take all capacity from all buckets
+	for _, bucket := range buckets {
+		_, _ = bucket.Limit(context.TODO())
+		_, _ = bucket.Limit(context.TODO())
+	}
+
+	// Wait for 2 seconds to check if it treats 0 TTL as infinite
+	clock.Sleep(2 * time.Second)
+
+	// Expect all buckets to be still filled
+	for name, bucket := range buckets {
+		s.Run(name, func() {
+			_, err := bucket.Limit(context.TODO())
+			s.Require().Equal(l.ErrLimitExhausted, err)
+		})
+	}
+}
+
+func (s *LimitersTestSuite) TestLeakyBucketTTLExpiration() {
+	clock := l.NewSystemClock()
+	buckets := s.leakyBuckets(1, time.Minute, time.Second, clock)
+
+	// Ignore in-memory bucket, as it has no expiration,
+	// ignore DynamoDB, as amazon/dynamodb-local doesn't support TTLs.
+	for k := range buckets {
+		if strings.Contains(k, "BucketInMemory") || strings.Contains(k, "BucketDynamoDB") {
```

Comment:
To avoid iterating over the buckets twice let's move this check to the for-loop below and skip InMemory and DynamoDB backend with `continue`

Reply:
Actually, I'm iterating 3 times:
* Ignore in-memory & DynamoDB buckets
* Fill all the buckets
* Check if the buckets are empty

If I skip this, I would either need to:
* **do this check in both loops:** which may be less clear
* **fill buckets and clear them in a single loop:** this would increase the test duration, as it works on a real clock - now it's `3 seconds`, while then it would be `number_of_buckets * 3 seconds` - 11 backends, 7 lockers = 231 seconds = ~4 minutes for this test + the same for TokenBucket one

Personally, I think it's the best way, but if you prefer I can adjust it in either way 👍